transcom/react-file-viewer
# B 21228 update pdfjs

### [B-21228](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1013105)

## Summary

- Upgrade node and pdfjs-dist.
- Adjust babel and jest for transpiling ESM modules into CJS.
- Miscellaneous changes due to transpiling adjustments (Fileviewer export syntax, add exports script to index.html, cheerio resolution, etc.)
- Addition of Playwright to simulate real browser environment
  - Jest was upgraded in an attempt to be able to use a real pdfjs import rather than a mock, but this did not work as it is still experimental within Jest.
  - Persisting the Jest upgrade for better future support.
- Added [polyfills](https://developer.mozilla.org/en-US/docs/Glossary/Polyfill) for future support of handling within the webpack bundle. This effort was halted due to discovery of [mymove](https://github.com/transcom/mymove) already handling polyfills.
  - Package upgrades persisted for future support.
- Workaround of https://github.com/transcom/mymove/security/dependabot/146 has been removed due to it being OBE by the pdfjs-dist upgrade.
- Upgraded `three` library and `.eslintrc` to provide current linter support.
- Added error catching.
- Adjusted dev webpack config to use chunks, this facilitates transpiling dependencies separately. This provided many benefits for use within the BundleAnalyzerPlugin.

Addition

### How to analyze

1. Clone locally
2. Run `yarn install`
3. Run `yarn serve:dev`
4. Preview webpack bundle on http://127.0.0.1:8888/
5. Preview app on http://localhost:8081/

### How to test

1. Clone locally
6. Run `yarn install`
7. Run `yarn test`
8. Run `yarn test:e2e`
9. Verify all tests pass
